### ISSUE TITLE
Docs only build not working after upgrading to 5.2.1

### ISSUE THREAD

**Issue by jgodi** [storybookjs/storybook#8148]
**Describe the bug**
Building with `--docs` no longer just builds the docs tab. It has the canvas tab as well.

**To Reproduce**
1. Build storybook with `--docs`
2. Load storybook
4. See canvas tab is loaded

**Expected behavior**
No tabs, just defaulting to the docs

**Screenshots**
![Screen Shot 2019-09-20 at 11 05 43 AM](https://user-images.githubusercontent.com/870799/65348689-9e5d4200-db96-11e9-87d7-469240ce7134.png)

**System:**
```    @storybook/addon-actions: ^5.2.1 => 5.2.1 
    @storybook/addon-docs: ^5.2.1 => 5.2.1 
    @storybook/addon-links: ^5.2.1 => 5.2.1 
    @storybook/addon-options: ^5.2.1 => 5.2.1 
    @storybook/addon-storysource: ^5.2.1 => 5.2.1 
    @storybook/addons: ^5.2.1 => 5.2.1 
    @storybook/cli: ^5.2.1 => 5.2.1 
    @storybook/react: ^5.2.1 => 5.2.1 ```

**Comment by shilman** [comment]
Upgraded storybook design system to 5.2.1 and not seeing this problem

https://github.com/storybookjs/design-system

Can you make a repro?

**Comment by jgodi** [comment]
Hmm, I will try to set it up again fresh and see what happens.

Side question, what is the best way to accomplish this. I want to have a header of docs written in MDX above my stories, then have my stories that are written in CSF loaded, then a footer of more docs at the bottom. Is this possible?

**Comment by shilman** [comment]
> I want to have a header of docs written in MDX above my stories, then have my stories that are written in CSF loaded, then a footer of more docs at the bottom.

This is possible right now, but we're going to make it easier in 5.3. Rather than working around the current structure, I'll try to release the better way of doing it in 5.3-alpha this coming week.

**Comment by elevatebart** [comment]
@jgodi we had this problem a lot on Chrome. 
It seems there is a light caching issue related to the LocalStorage.
Please clear both LocalStorage and CachedRessources and tell us how it works.

I hope this helps a little.

**Comment by jgodi** [comment]
> > I want to have a header of docs written in MDX above my stories, then have my stories that are written in CSF loaded, then a footer of more docs at the bottom.
> 
> This is possible right now, but we're going to make it easier in 5.3. Rather than working around the current structure, I'll try to release the better way of doing it in 5.3-alpha this coming week.

Thats awesome, let me know when that is available. I will definitely test it out right away.

**Comment by jgodi** [comment]
@elevatebart Tried this and even tried a different browser and still I get canvas and docs. I will wait until the new 5.3-alpha and try again when I redo our stories. Thanks!

**Comment by shilman** [comment]
@jgodi releasing a 5.3-alpha but have not made the fix yet. will try to get to it in the next day or two!

**Comment by jgodi** [comment]
Tried everything today to get the docs only build working. It builds with Canvas/Docs tabs and defaults to the Docs tab, but before we had only docs mode deployed. Is this still a thing?

**Comment by shilman** [comment]
@jgodi I'm not sure I understand your question. As far as I know, `--docs` mode is still working.

**Comment by jgodi** [comment]
![Screen Shot 2019-09-24 at 5 05 47 PM](https://user-images.githubusercontent.com/870799/65558998-afd38080-deed-11e9-82a9-654a21e51826.png)

If you look at the screenshot above, you see the Canvas tab and Docs tab. This was build with us being on version 5.2.1 and using `build-storybook --docs`.

When we were on 5.2.0-alpha.36 and we built using `build-storybook --docs`, then the output would be no tabs and just loaded the docs only. 

Is there a way to go back to having no tabs displayed and be in only docs mode?

**Comment by shilman** [comment]
@jgodi Do you have a reproduction? It's working in our repos, so I'd need to take a look at a broken repo to debug what's going on, and I don't have one.

**Comment by jgodi** [comment]
@shilman Repro here in this repo: https://github.com/jgodi/storybook-test

Even tried with v5.3.0-alpha.2 of @storybook/addon-docs

**Comment by shilman** [comment]
@jgodi remove the showPanel option: https://github.com/jgodi/storybook-test/blob/master/.storybook/config.js#L8

See also: https://github.com/storybookjs/storybook/issues/8063

**Comment by jgodi** [comment]
Ah, thanks. I searched and searched before opening the issue too. Glad it was an easy fix. Thank you for looking into it and helping out.

**Comment by jgodi** [comment]
> > I want to have a header of docs written in MDX above my stories, then have my stories that are written in CSF loaded, then a footer of more docs at the bottom.
> 
> This is possible right now, but we're going to make it easier in 5.3. Rather than working around the current structure, I'll try to release the better way of doing it in 5.3-alpha this coming week.

@shilman any update on this part? Would love to try this out.

**Comment by shilman** [comment]
@jgodi Not yet, I'll try to get something out this week!

**Comment by tuncaulubilge** [comment]
I'm still getting this issue on 6.3.2. `start-storybook --docs` does work as expected, with flattened list of components, and the canvas tab hidden etc, but `build-storybook --docs` produces a regular storybook, without the doc mode enabled.

**Comment by shilman** [comment]
@tuncaulubilge do you have a reproduction repo you can share? Or you can create one by running `npx sb@next repro`, following the instructions, and linking it in your issue description? We prioritize issues with reproductions over those without. Thank you! 🙏

**Comment by tuncaulubilge** [comment]
@shilman I was able to reproduce it with a fresh CRA app and `npx sb init`
See the repo: https://github.com/tuncaulubilge/storybook-docs-repro

Just run `yarn build-storybook:docs` and `npx server storybook-static` to see the results. The UI is quite different than what you'd see with `yarn storybook:docs`

**Comment by tuncaulubilge** [comment]
@shilman Sorry to be the guy that keeps asking "Are we there yet?", but is there a roadmap on this one. Currently there are no workarounds on this as far as I'm aware, so we are trying to assess whether to wait for Storybook Docs, or build a regular storybook until the issue is fixed.

**Comment by ABXlink** [comment]
This is still an issue in `6.3.12`.

**Comment by kylemh** [comment]
I'm seeing this on the latest version of Storybook.


https://user-images.githubusercontent.com/9523719/147345329-4e5eba66-74e4-44cc-949a-57e2802f3554.mov

**Comment by kylemh** [comment]
Maybe this can be closed for #15933

**Comment by shilman** [comment]
thanks @kylemh ! closing as dupe to #15933
